ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/main.yml

```diff
@@ -48,6 +49,16 @@
   when: Domain == "adoptopenjdk.net" and ansible_distribution == "MacOSX"
   tags: hostname,adoptopenjdk
 
+- name: Set hostname to inventory_hostname (Solaris)
+  shell: "{{ item }}"
+  with_items:
+    - echo "{{ inventory_hostname }}" > /etc/nodename
+    - echo "{{ inventory_hostname }}" > /etc/hostname.e1000g0
```

Comment:
my understanding is that on nearly all Solaris systems the default is `e1000g0`. I'm not very happy with this approach for setting the hostname in general - it seems very hacky

Reply:
Certainly on Solaris 10 I've had interface names of `bge0` although I'd suggest adding a comment to this saying that it may not work if the interface name is not `e1000g0`, or even `TODO: Allow other interfaces` above this section

Reply:
updated